### PR TITLE
Fix incomplete battery list in Global.batteries.model

### LINKED ISSUE #1224: List of batteries from overview drilldown is incomplete
Clicking on the battery from the overview page shows a list of batteries in the system, if there is more than one battery.

The battery list is generated from `com.victronenergy.battery.*` but this is not sufficient. Some batteries are missing; see [this](https://victrondevelopment.slack.com/archives/C04RYU0UMDE/p1718001991024539) slack thread.

### PR DESCRIPTION
**Pull request by blammit**
Show a complete list of batteries when the battery overview widget is clicked. The list should come from com.victronenergy.system/~AvailableBatteries~Batteries rather than com.victronenergy.battery.*.

Fixes #1224

### CODE REVIEW

**Comment by blammit** [comment]
This needs some fixes for the dbus case. Will update.

**Comment by blammit** [comment]
> This needs some fixes for the dbus case. Will update.

Updated now, ready to review.

**Comment by blammit** [comment]
Updated to fetch batteries from /Batteries instead of /AvailableBatteries. Ready to review.

**Comment by blammit** [comment]
@DanielMcInnes could you review this one please?

**Comment by DanielMcInnes** [comment]
I see this error on startup when connecting to my local cerbo:
qrc:/qt/qml/Victron/VenusOS/data/Batteries.qml:87: TypeError: Cannot read property 'model' of undefined

File: src/basedevicemodel.cpp [hunk 1]
```diff
@@ -167,6 +167,16 @@ bool BaseDeviceModel::removeDevice(const QString &serviceUid)
 	return false;
 }
 
+void BaseDeviceModel::intersect(const QStringList &serviceUids)
+{
+	for (int i = 0; i < m_devices.count(); ++i) {
```
Comment (DanielMcInnes) [inline src/basedevicemodel.cpp:172]: I think you need to iterate over m_devices in reverse if you might be removing elements

Comment (blammit) [reply src/basedevicemodel.cpp]: Oops, yes. Fixed

File: data/Batteries.qml [hunk 2]
```diff
@@ -64,5 +64,53 @@ QtObject {
 		}
 	}
 
+	readonly property VeQuickItem _batteries: VeQuickItem {
+		uid: Global.system.serviceUid + "/Batteries"
+		onValueChanged: {
+			let i
+			if (!isValid) {
+				if (Global.batteries) {
+					for (i = Global.batteries.model.count - 1; i >= 0; ++i) {
```
[the remark this answers is not in the document]
Comment (blammit) [reply data/Batteries.qml]: The objects will just stick around, since they are not automatically destroyed when removed from the model

**Review by DanielMcInnes — APPROVED** [review]

**Pull request merged**